### DISCUSSION TITLE
jade view engine indentation error in mdn express tutorial with solution.

### DISCUSSION THREAD

**Discussion by harunfr** [mdn/mdn-community#574] [Content]
https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/Displaying_data/LocalLibrary_base_template
When I copied and paste this code to layout.jade file and save, then run it, I got an indentation error. 
I solved this problem by resetting indentations.
More imperative solution: click on head press end and then press delete until no whitespace, then hit tab to times to give indent. Do this until you come to body tag.

Mozilla please solve this problem in original layout.jade file. Thanks.

**Comment by hamishwillee** [comment]
There is no layout.jade file in this tutorial. Do you mean layout.pug, and have you selected the pug layout engine?

Note that I'm not saying that this might not be a problem for pug as well, but I don't want to spend the time debugging this if you are working the wrong layout engine than the tutorial.

Can you paste in the error you got?

**Comment by pepelsbey** [comment]
Closing as outdated/irrelevant.
